torchmultimodal/modules/layers/attention.py

```diff
@@ -12,6 +13,67 @@
 from torchmultimodal.utils.common import shift_dim
 
 
+class AxialAttentionBlock(nn.Module):
+    """Computes multihead axial attention across all dims of the input.
+
+    Axial attention is an alternative to standard full attention, where instead
+    of computing attention across the entire flattened input, you compute it for
+    each dimension. To capture the global context that full attention does, stacking
+    multiple axial attention layers will allow information to propagate among the
+    multiple dimensions of the input. This enables attention calculations on high
+    dimensional inputs (images, videos) where full attention would be computationally
+    expensive and unfeasible. For more details, see "Axial Attention in
+    Multidimensional Transformers" (Ho et al. 2019) and CCNet (Huang et al. 2019).
+
+    Follows implementation by VideoGPT:
+    https://github.com/wilson1yan/VideoGPT/blob/master/videogpt/vqvae.py
+
+    Args:
+        n_dims (int): dimensionality of input data, not including batch or channel dims
+        qkv_dim (int): dimensionality of linear projections Wq, Wk, and Wv in attention
+        n_head (int): number of heads in multihead attention. Must divide into qkv_dim
+                      evenly
+
+    Inputs:
+        x (Tensor): a [b, c, d1, ..., dn] tensor, where c == qkv_dim
+    """
+
+    def __init__(self, n_dims: int, qkv_dim: int, n_head: int) -> None:
+        super().__init__()
+        self.qkv_dim = qkv_dim
+        self.mha_attns = nn.ModuleList(
+            [
+                MultiHeadAttention(
+                    shape=tuple(
+                        repeat(0, n_dims)
+                    ),  # dummy value for shape since we are not using causal
+                    dim_q=qkv_dim,
+                    dim_kv=qkv_dim,
+                    n_head=n_head,
+                    n_layer=1,
+                    causal=False,
+                    attn_module=AxialAttention(d),
```

Comment:
Not necessarily a comment for this PR, but it might be useful to provide more info on the in/out dims expected for attn_module in the docstring for MultiHeadAttention. Though your usage here provides a good example, being explicit about the APIs will make it even easier for any users that want to plug in their own attention modules.